app/helpers/product_helper.rb

```diff
@@ -23,4 +23,18 @@ def products_except(product_id)
       prod[:product_id] == product_id
     end
   end
+
+  def format_product(product)
+    product[:product_quantity] = safe_integer_cast(product[:product_quantity])
+    product[:lead_time] = safe_integer_cast(product[:lead_time])
+    product
+  end
+
+private
+
+  def safe_integer_cast(number)
+    Integer(number)
+  rescue ArgumentError, TypeError
+    nil
```

Comment:
I wonder if we want to raise an error here, and get the user to give us something we can coerce into an integer?

Otherwise, we might end up putting records into the DB without product lead time or quantity provided.

Reply:
We do validate earlier and re-request from the user.
Then we cast the responses once all validation passes. So, it **should** be impossible to get an error at this point.